src/app/pages/transaction-request/transaction-request.tsx

```diff
@@ -51,14 +57,23 @@ function TransactionRequestBase() {
 
   const generateUnsignedTx = useGenerateUnsignedStacksTransaction();
   const stxAddress = useCurrentStacksAccountAddress();
+
   const { filteredBalanceQuery } = useStxCryptoAssetBalance(stxAddress);
   const availableUnlockedBalance = filteredBalanceQuery.data?.availableUnlockedBalance;
 
   const { data: nextNonce, status: nonceQueryStatus } = useNextNonce(stxAddress);
-  const canSubmit = filteredBalanceQuery.status === 'success' && nonceQueryStatus === 'success';
+
+  const { isVerifying: isVerifyingSbtcSponsorship, result: sbtcSponsorshipEligibility } =
+    useCheckSbtcSponsorshipEligible(unsignedTx, nextNonce, stxFees);
```

Comment:
Concerned if you check the unsigned tx here, instead of in the onSubmit, then you won't be checking any form value changes? I believe at this point, the unsigned tx is just using defaults to calc fees?

Reply:
@fbwoolf The sponsored unsigned tx adopts the default/moderate fee based on the transaction request data and then disables selection on the FeeForm component (`sponsored=true`), so fees shouldn't be changeable. Also the verification check should run again (and generates a new unsigned tx) if the user changes account on the confirmation screen. Those were the only updatable values I saw but lmk if there are other possible form changes I missed.

Reply:
Yep, I see now that I've pulled the branch and looked at it closer. 👍